controllers/controllers/services/cfservicebinding_controller.go

```diff
@@ -36,21 +36,29 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+//counterfeiter:generate -o fake -fake-name VCAPServicesSecretBuilder . VCAPServicesSecretBuilder
+type VCAPServicesSecretBuilder interface {
+	BuildVCAPServicesEnvValue(context.Context, *korifiv1alpha1.CFApp) (string, error)
+}
+
 // CFServiceBindingReconciler reconciles a CFServiceBinding object
 type CFServiceBindingReconciler struct {
 	client.Client
-	Scheme *runtime.Scheme
-	Log    logr.Logger
+	Scheme  *runtime.Scheme
+	Log     logr.Logger
+	Builder VCAPServicesSecretBuilder
 }
 
-func NewCFServiceBindingReconciler(client client.Client, scheme *runtime.Scheme, log logr.Logger) *CFServiceBindingReconciler {
-	return &CFServiceBindingReconciler{Client: client, Scheme: scheme, Log: log}
+func NewCFServiceBindingReconciler(client client.Client, scheme *runtime.Scheme, log logr.Logger, builder VCAPServicesSecretBuilder) *CFServiceBindingReconciler {
+	return &CFServiceBindingReconciler{Client: client, Scheme: scheme, Log: log, Builder: builder}
 }
 
 const (
-	BindingSecretAvailableCondition   = "BindingSecretAvailable"
-	ServiceBindingGUIDLabel           = "korifi.cloudfoundry.org/service-binding-guid"
-	ServiceCredentialBindingTypeLabel = "korifi.cloudfoundry.org/service-credential-binding-type"
+	BindingSecretAvailableCondition      = "BindingSecretAvailable"
```

Comment:
I might be overlooking something but I do not see the point of having the `BindingSecretAvailableCondition` and `VCAPServicesSecretAvailableCondition` conditions at all. Looking at the code, I can see them only being set, I would imagine that someone should be looking at them in order to justify their existence.
Or are they around for the sake of traceability?

Reply:
I believe those status are for debugging through the K8s API. Regarding `VCAPServicesSecretAvailableCondition` not ever being set to `true`, you're right and I think the one on line 230 should be `VCAPServicesSecretAvailableCondition` and not `BindingSecretAvailableCondition` so I fixed that.